File: src/fund_analyzer/api/schemas/risk.py
"""风险评估相关 API 数据模型。"""
from __future__ import annotations

from pydantic import BaseModel


class RiskSubmitRequest(BaseModel):
    """风险问卷提交请求体。"""

    answers: dict[str, int]


class RiskProfileResponse(BaseModel):
    """风险评估结果响应。"""

    id: int | None = None
    risk_level: int
    risk_label: str
    core_ratio: int
    satellite_ratio: int
    assessment_date: str
    answers: dict | None = None

File: web/src/types/risk.ts
/** 问卷选项 */
export interface QuestionOption {
  label: string
  score: number
}

/** 问卷问题 */
export interface Question {
  id: string
  text: string
  options: QuestionOption[]
}

/** 风险评估档案 */
export interface RiskProfile {
  id: number | null
  risk_level: number
  risk_label: string
  core_ratio: number
  satellite_ratio: number
  assessment_date: string
  answers: Record<string, number> | null
}
